compilation broken on armv7 linux

rpi3 / Raspbian Jessie / gcc 4.9

```
make -f Makefile.libretro platform=armvhardfloatunix TILED_RENDERING=1 HAVE_NEON=1
g++ -c -o src/sound.o src/sound.cpp -DGIT_VERSION=\"" 23e7f0d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
g++ -c -o src/memory.o src/memory.cpp -DGIT_VERSION=\"" 23e7f0d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
g++ -c -o src/gba.o src/gba.cpp -DGIT_VERSION=\"" 23e7f0d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
g++ -c -o src/system.o src/system.cpp -DGIT_VERSION=\"" 23e7f0d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
g++ -c -o libretro/libretro.o libretro/libretro.cpp -DGIT_VERSION=\"" 23e7f0d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
cc -c -o libretro-common/memalign.o libretro-common/memalign.c -O2 -DNDEBUG  -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
cc    -c -o src/neon_memcpy.o src/neon_memcpy.S
cc -o vba_next_libretro.so -shared -Wl,--no-undefined ./src/sound.o ./src/memory.o ./src/gba.o ./src/system.o ./libretro/libretro.o ./libretro-common/memalign.o ./src/neon_memcpy.o  
./src/sound.o: In function `Blip_Buffer::clock_rate_factor(long) const':
sound.cpp:(.text+0x1c64): undefined reference to `floor'
./src/sound.o: In function `_GLOBAL__sub_I_sound.cpp':
sound.cpp:(.text.startup+0xbc): undefined reference to `__cxa_end_cleanup'
./src/sound.o:(.ARM.extab.text.startup+0x0): undefined reference to `__gxx_personality_v0'
./src/gba.o: In function `CPUSoftwareInterrupt(int)':
gba.cpp:(.text+0x9363c): undefined reference to `pow'
gba.cpp:(.text+0x95938): undefined reference to `sqrt'
collect2: error: ld returned 1 exit status
Makefile.libretro:413: recipe for target 'vba_next_libretro.so' failed
make: *** [vba_next_libretro.so] Error 1
```

bisected to 23e7f0d5264db7322d577d929d30e97428c99ea0

successful compilation looks like

```
make clean; make -f Makefile.libretro platform=armvhardfloatunix TILED_RENDERING=1 HAVE_NEON=1
rm -f ./src/sound.o ./src/memory.o ./src/gba.o ./src/system.o ./libretro/libretro.o ./libretro-common/memalign.o 
rm -f  ./libretro-common/memalign.o ./libretro-common/rthreads/rthreads.o ./src/thread.o ./src/neon_memcpy.o
rm -f vba_next_libretro.so
g++ -c -o src/sound.o src/sound.cpp -DGIT_VERSION=\"" cb98f5d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
g++ -c -o src/memory.o src/memory.cpp -DGIT_VERSION=\"" cb98f5d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
g++ -c -o src/gba.o src/gba.cpp -DGIT_VERSION=\"" cb98f5d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
g++ -c -o src/system.o src/system.cpp -DGIT_VERSION=\"" cb98f5d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
g++ -c -o libretro/libretro.o libretro/libretro.cpp -DGIT_VERSION=\"" cb98f5d"\" -O2 -DNDEBUG -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
cc -c -o libretro-common/memalign.o libretro-common/memalign.c -O2 -DNDEBUG  -fPIC -D__LIBRETRO__ -DINLINE=inline -DHAVE_HLE_BIOS -DFRONTEND_SUPPORTS_RGB565 -DUSE_CHEATS -DTILED_RENDERING  -marm -mfloat-abi=hard -DARM -DHAVE_STDINT_H -DHAVE_INTTYPES_H  -I. -I./libretro-common/include
cc    -c -o src/neon_memcpy.o src/neon_memcpy.S
g++ -o vba_next_libretro.so -shared -Wl,--no-undefined ./src/sound.o ./src/memory.o ./src/gba.o ./src/system.o ./libretro/libretro.o ./libretro-common/memalign.o ./src/neon_memcpy.o  
```

this fixes the linking - g++ is needed to link in this case rather than CC
```diff
diff --git a/Makefile.libretro b/Makefile.libretro
index 72822b4..5cb1605 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -403,7 +403,7 @@ ifneq (,$(findstring msvc,$(platform)))
        LINKOUT = -out:
        LD = link.exe
 else
-       LD = $(CC)
+       LD = $(CXX)
 endif
 
 $(TARGET): $(OBJECTS)
```

will submit a PR

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/48167665-compilation-broken-on-armv7-linux?utm_campaign=plugin&utm_content=tracker%2F1027743&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1027743&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -403,7 +403,7 @@ ifneq (,$(findstring msvc,$(platform)))
 	LINKOUT = -out:
 	LD = link.exe
 else
-	LD = $(CC)
+	LD = $(CXX)
 endif
 
 $(TARGET): $(OBJECTS)

